### ISSUE TITLE
Variable is uneccessary

### ISSUE THREAD

**Issue by erikvader** [erikvader/MiniZinc-linter#8]

**Comment by erikvader** [comment]
There was someone that mentioned that new students liked to initialize arrays with a variable in a bad way somehow. I don't remember how exactly (my notes were vague).

**Comment by GustavBjordal** [comment]
I think this was that we want to avoid people using variables to express parameters and set their values via constraints.
So for example the following is actually common among beginners:
```
array[1..n] of var int: param;
constraint forall(i in 1..n)(param[i] = i);
```
or the slightly better but still bad
```
array[1..n] of var int: param = [i | i in 1..n];
```

The correct version is:
```
array[1..n] of int: param = [i | i in 1..n];
```

**Comment by erikvader** [comment]
Oh okay! So it is basically checking whether a variable (or an array of them) is actually a parameter? This could probably be merged with #4 or something.

**Comment by Pierre-Flener** [comment]
Yes, this sounds like #4  indeed.  Highest priority in any case!
